### PR TITLE
feat(usesandpack): reset-file and reset-all-files CSB-829

### PR DESCRIPTION
**Pull request by danilowoz**
It implements the ability to reset the files for the original values from `props`. The following methods were added to the context:
| Method | Description |
| - | - |
| `resetAllFiles` | Reset all files for all paths               |
| `resetFile`     | Reset the code for a given path             |

CSB-829

### CODE REVIEW

File: sandpack-react/src/contexts/sandpackContext.tsx [hunk 1]
```diff
@@ -541,21 +555,23 @@ class SandpackProvider extends React.PureComponent<
...
-      updateCurrentFile: this.updateCurrentFile,
-      updateFile: this.updateFile,
-      runSandpack: this.runSandpack,
-      registerBundler: this.registerBundler,
-      unregisterBundler: this.unregisterBundler,
-      dispatch: this.dispatchMessage,
-      listen: this.addListener,
       deleteFile: this.deleteFile,
-      lazyAnchorRef: this.lazyAnchorRef,
+      dispatch: this.dispatchMessage,
```
Comment (danilowoz) [inline sandpack-react/src/contexts/sandpackContext.tsx:560]: Sort methods

**Review by alexnm — APPROVED** [review]
Nice 💯 !

File: sandpack-react/README.md [hunk 2]
```diff
@@ -543,6 +543,22 @@ const CustomRefreshButton = () => {
...
+| Method          | Description                                 |
+| --------------- | ------------------------------------------- |
+| `closeFile`     | Close the given path in the editor          |
+| `deleteFile`    | Delete the given path in the editor         |
+| `dispatch`      | Sends a message to the bundler              |
+| `listen`        | Listens for messages from the bundler       |
+| `openFile`      | Open the given path in the editor           |
+| `resetAllFiles` | Reset all files for all paths               |
+| `resetFile`     | Reset the code for a given path             |
+| `setActiveFile` | Set as active the file for a given path     |
```
Comment (codebryo) [inline sandpack-react/README.md:559]: `Set a specific file as active in a given path` would be better.

File: sandpack-react/README.md [hunk 3]
```diff
@@ -543,6 +543,22 @@ const CustomRefreshButton = () => {
...
+| `updateFile`    | Give a path and a code, it updates the file |
```
Comment (codebryo) [inline sandpack-react/README.md:560]: `Update the content of a file in a given path`

**Review by codebryo — COMMENTED** [review]
Overall looks good - maybe on the docs we should add a API example as well or explain how to use it.

`closeFile('path/to/file')` or `updateFile('path', 'content to replace')` 

Unless path needs to a speficif object or such :)

**Pull request merged**